waltz-tools/src/main/java/com/wepay/waltz/tools/client/ClientCli.java

```diff
@@ -91,14 +94,14 @@ private ClientCli(String[] args,  boolean useByTest) {
 
         private static final List<PartitionLocalLock> LOCKS = Collections.singletonList(new PartitionLocalLock(LOCK_NAME, LOCK_ID));
 
-        private final Map<Integer, ConcurrentHashMap<Integer, Long>> clientHighWaterMarkMap;
-        private final List<String> uncaughtExceptions;
+        protected final Map<Integer, ConcurrentHashMap<Integer, Long>> clientHighWaterMarkMap;
```

Comment:
Instead of making this class and few of its variables protected, why not create a base class that contain the command member functions and have `Validate`, `Producer` and `Consumer` class extend on top of it.